pkg/kademlia/routing.go

```diff
@@ -321,3 +322,77 @@ func (rt *RoutingTable) ConnSuccess(ctx context.Context, node *pb.Node) {
 		zap.L().Debug("connection success error:", zap.Error(err))
 	}
 }
+
+func splitBucket(bIDs []bucketID, bitDepth int) ([]bucketID, []bucketID) {
+	var b0 []bucketID
+	var b1 []bucketID
+
+	for _, bID := range bIDs {
+		byteDepth := bitDepth / 8
+		bitOffset := bitDepth % 8
+		power := uint(7 - bitOffset)
+		bitMask := byte(1 << power)
+		b := bID[byteDepth]
+		if b&bitMask > 0 {
+			b1 = append(b1, bID)
+		} else {
+			b0 = append(b0, bID)
+		}
+	}
+	return b0, b1
+}
+
+// BufferedGraph prints the routing table graph as a dot graph in the specified buffer
+func (rt *RoutingTable) BufferedGraph(buf *bytes.Buffer) {
+	buf.Write([]byte("digraph{\nnode [shape=box];edge [dir=none];\n"))
```

Comment:
maybe, ranking left-to-right, might make the output look nicer

Reply:
Here are two examples: default and with left-to-right ranking. I don't know which one is better.


![default](https://user-images.githubusercontent.com/17983871/56358100-ed650c80-61dd-11e9-97c5-65048eaaf3d7.png)
![left-to-right](https://user-images.githubusercontent.com/17983871/56358104-f1912a00-61dd-11e9-874d-ef3150f4e01b.png)

What do you think?

Reply:
Both don't look that good :D. Maybe if the font-size was smaller for the node-ids and addresses both would look better.

Reply:
The first layout looks more intuitive to me. I am used to looking at binary trees from top to down, not from left to right.